lib/middlewares/bulk.js

```diff
@@ -0,0 +1,57 @@
+"use strict"
+
+var _ = require('lodash')
+var RabbitQueue = require('./rabbit_queue')
+var amqp = require('amqplib')
+var co = require('co')
+
+class Bulk extends RabbitQueue {
+  constructor(options) {
+    super(options)
+
+    this.messages = this.service.subscribedBulks
+    this.defineMandatoryDependency('in_queue', ':service_in')
+  }
+
+  getQueueNameFor(msgType) {
+    return `${this.service.name}_${msgType}_bulk`
+  }
+
+  *initBulk(bulk) {
+    bulk.channel = yield this.connection.createChannel()
+    yield bulk.channel.assertQueue(this.getQueueNameFor(bulk.type))
+  }
+
+  *initialize() {
+    this.connection = yield amqp.connect(this.amqpUrl)
+    yield this.messages.map(_ => this.initBulk(_))
+  }
+
+  *enqueueBulk(bulk, buffer) {
+    var msg = buffer.map(_ => JSON.parse(_.content.toString()))
+    yield this.getDependency(':service_in').enqueue(msg)
+    buffer.forEach(_ => bulk.channel.ack(_))
+  }
+
+  *packBulks(bulk) {
```

Comment:
We should talk about the possibility of adding a timeout here since the bulking is per-process, would be really easy with the current implementation